eth_bloom/bloom.py

```diff
@@ -2,54 +2,64 @@
 
 import numbers
 import operator
+from typing import (
+    Iterable,
+    List,
+    Type,
+    TypeVar,
+    Union,
+)
 
 from eth_hash.auto import keccak as keccak_256
 
 
-def get_chunks_for_bloom(value_hash):
+def get_chunks_for_bloom(value_hash: bytes) -> Iterable[bytes]:
     yield value_hash[:2]
     yield value_hash[2:4]
     yield value_hash[4:6]
 
 
-def chunk_to_bloom_bits(chunk):
+def chunk_to_bloom_bits(chunk: bytes) -> int:
     high, low = bytearray(chunk)
     return 1 << ((low + (high << 8)) & 2047)
 
 
-def get_bloom_bits(value):
+def get_bloom_bits(value: bytes) -> Iterable[int]:
     value_hash = keccak_256(value)
     for chunk in get_chunks_for_bloom(value_hash):
         bloom_bits = chunk_to_bloom_bits(chunk)
         yield bloom_bits
 
 
+T = TypeVar('T', bound='BloomFilter')
+
+
 class BloomFilter(numbers.Number):
     value = None
 
-    def __init__(self, value=0):
+    def __init__(self, value: int=0) -> None:
         self.value = value
 
-    def __int__(self):
+    def __int__(self) -> int:
         return self.value
 
-    def add(self, value):
+    def add(self, value: bytes) -> None:
         if not isinstance(value, bytes):
             raise TypeError("Value must be of type `bytes`")
         for bloom_bits in get_bloom_bits(value):
             self.value |= bloom_bits
 
-    def extend(self, iterable):
+    def extend(self, iterable: List[bytes]) -> None:
```

Comment:
Here as well as in other locations, I think we can use `Sequence` instead of `List` as is outlined here: https://github.com/ethereum/snake-charmers-tactical-manual/blob/master/style-guide.md#choosing-the-right-type

Reply:
Even `Iterable[bytes]` should work here as we are really just iterating over them.